src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/AsyncWriteProxyEx.cs

```diff
@@ -388,13 +388,13 @@ public static Task<T> AskEx<T>(this ICanTell self, Func<IActorRef, object> messa
             return self.AskEx<T>(messageFactory, null, cancellationToken);
         }
 
-        public static Task<T> AskEx<T>(this ICanTell self, Func<IActorRef, object> messageFactory, TimeSpan? timeout, CancellationToken cancellationToken)
+        public static async Task<T> AskEx<T>(this ICanTell self, Func<IActorRef, object> messageFactory, TimeSpan? timeout, CancellationToken cancellationToken)
         {
             IActorRefProvider provider = ResolveProvider(self);
             if (provider == null)
                 throw new ArgumentException("Unable to resolve the target Provider", nameof(self));
 
-            return AskEx(self, messageFactory, provider, timeout, cancellationToken).CastTask<object, T>();
+            return (T)await AskEx(self, messageFactory, provider, timeout, cancellationToken);
```

Comment:
We don't need this CastTask at all. All we need to do here is to take await and cast the value from object to T.

The rest will be managed by the .net itself. All the exceptions will be propagated as is.